doc/conftest.py

```diff
@@ -129,6 +137,16 @@ def pytest_runtest_setup(item):
     elif fname.endswith("statistical_inference/unsupervised_learning.rst"):
         setup_unsupervised_learning()
 
+    rst_files_requiring_matplotlib = [
+        "modules/partial_dependence.rst",
+        "modules/tree.rst",
+        "tutorial/statistical_inference/settings.rst",
+        "tutorial/statistical_inference/supervised_learning.rst",
+    ]
+    for each in rst_files_requiring_matplotlib:
+        if fname.endswith(each):
```

Comment:
The following should do:

```suggestion
        if fname.endswith(each.replace("/", os.path.sep)):
```

Reply:
I ended up fixing this in https://github.com/scikit-learn/scikit-learn/pull/20356/commits/2c5d6f962ab87e0c8753f49a04b732b79356b820 since this problem happens for the other filename tests in this function.